src/system_database.ts

```diff
@@ -1270,10 +1247,31 @@ export class PostgresSystemDatabase implements SystemDatabase {
       query = query.offset(input.offset);
     }
 
-    const rows = await query.select(`${DBOSExecutor.systemDBSchemaName}.workflow_status.workflow_uuid`);
-    const workflowUUIDs = rows.map((row) => (row as { workflow_uuid: string }).workflow_uuid);
+    const rows = await query;
+    return rows.map(PostgresSystemDatabase.#mapWorkflowStatus);
+  }
+
+  static #mapWorkflowStatus(row: workflow_status & workflow_inputs): WorkflowStatusInternal {
     return {
-      workflowUUIDs: workflowUUIDs,
+      workflowUUID: row.workflow_uuid,
+      status: row.status,
+      workflowName: row.name,
+      output: row.output ? row.output : null,
+      error: row.error ? row.error : null,
+      workflowClassName: row.class_name ?? '',
+      workflowConfigName: row.config_name ?? '',
+      queueName: row.queue_name,
+      authenticatedUser: row.authenticated_user,
+      assumedRole: row.assumed_role,
+      authenticatedRoles: JSON.parse(row.authenticated_roles) as string[],
+      request: JSON.parse(row.request) as HTTPRequest,
```

Comment:
Can we stop returning the `request` directly? I'll be working on removing its storage soon

Reply:
I don't have a strong opinion on this, but might be better handled in a follow up PR

Reply:
yeah I'll handle it in a future PR no worries